### PR TITLE
Feature/under menu buttons

### PR DESCRIPTION
**Pull request by breakdancingcat**
* https://waynedotedu.tpondemand.com/entity/76671-base-undermenu-promotion-improvement

### Overview
* Default button style is now grey-lighter background with green text
* Created a series of button variants.
* Ideally these wouldn't all be mixed together on the same page, the user would select one style
* Each variant is selected by dropdown

### Button variants
* No option selected
    * Default button style, grey-lighter bg and green text
    * One or two lines of text if excerpt is filled in
* Default dark
    * Green gradient background
    * White text
    * One or two lines of text if excerpt is filled in
* Icon dark theme
    * Green gradient background
    * White text
    * Upload white svg icon as 'secondary_filename', it appears to the left of the text
    * One or two lines of text if excerpt is filled in
* Icon light theme
    * Grey-lighter background
    * Black text
    * Upload black svg icon as 'secondary_filename', it appears to the left of the text
    * One or two lines of text if excerpt is filled in
* Bg image light
    * Filename background 2x medium quality jpg, darker image
    * Dark text
* Bg image dark
    * Filename background 2x medium quality jpg, lighter image
    * Light text
* SVG overlay light
    * Filename background 2x medium quality jpg, lighter image
    * Semi-transparent light overlay for contrast
    * Upload svg overlay as 'secondary_filename', 360 x 131 px with dark text
* SVG overlay dark
    * Filename background 2x medium quality jpg, darker image
    * Semi-transparent dark overlay for contrast
    * Upload svg overlay as 'secondary_filename', 360 x 131 px with light text

screencapture-base-wayne-local-styleguide-undermenu-2018-09-07-18_09_56

### CODE REVIEW

File: resources/views/components/image-button-list.blade.php [hunk 1]
```diff
@@ -13,6 +13,9 @@
                 @else
                     @image($image['relative_url'], $image['title'])
                 @endif
+            @elseif(!empty($image['relative_url']) && !empty($image['secondary_image']))
+                <img src="/promos/{{ $image['promo_group_id'] }}/{{ $image['secondary_image']}}">
```
Comment (nickdenardis) [inline resources/views/components/image-button-list.blade.php]: Should this be updated with the new `secondary_relative_url` value?

Comment (breakdancingcat) [reply resources/views/components/image-button-list.blade.php]: I think we're deleting this file, I didn't mean to commit this change

**Comment by breakdancingcat** [comment]
@robertvrabel I removed image button list and image button list rotate. Should we correct the menu id's? I also need to remove the test for that component and add one for this.

**Comment by robertvrabel** [comment]
@breakdancingcat Yea lets adjust the IDs so they are in order. If you need assistance with any of the changes to the tests let me know!

**Comment by breakdancingcat** [comment]
@robertvrabel @nickdenardis I think this is good to go now!

File: resources/views/components/button-svg-overlay-light.blade.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{--
+    $button => array // [['title', 'link', 'excerpt', 'relative_url', 'secondary_relative_url']]
+--}}
+
+@if(!empty($button['link']) && !empty($button['relative_url']) && !empty($button['secondary_relative_url']))
+    <a href="{{ $button['link'] }}" class="block min-w-full relative rounded bg-cover bg-grey-lighter" style="padding-top: 36.39%; background-image: url('{{ $button['relative_url'] }}'); ">
+        <div class="absolute pin p-4 rounded bg-white opacity-65"></div>
+        <div class="absolute min-w-full pin rounded">
+            <img src="{{ $button['secondary_relative_url']}}">
```
Comment (nickdenardis) [inline resources/views/components/button-svg-overlay-light.blade.php]: This needs an `alt` attribute with the text that should be read to a screen reader. Likely the `title`.

File: resources/views/components/button-svg-overlay-dark.blade.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{{--
+    $button => array // [['title', 'link', 'relative_url', 'secondary_relative_url']]
+--}}
+
+@if(!empty($button['link']) && !empty($button['relative_url']) && !empty($button['secondary_relative_url']))
+    <a href="{{ $button['link'] }}" class="block min-w-full relative rounded bg-cover bg-green-darkest" style="padding-top: 36.39%; background-image: url('{{ $button['relative_url'] }}'); ">
+        <div class="absolute pin p-4 rounded bg-green-darkest opacity-65"></div>
+        <div class="absolute min-w-full pin rounded">
+            <img src="{{ $button['secondary_relative_url']}}">
```
Comment (nickdenardis) [inline resources/views/components/button-svg-overlay-dark.blade.php]: This also needs an `alt` attribute with the text that should be read to a screen reader. Likely the `title`.

**Pull request closed without merging**